ts/defs.bzl

```diff
@@ -439,6 +494,7 @@ def ts_project(
         tsc = tsc,
         tsc_worker = tsc_worker,
         transpile = -1 if not transpiler else int(transpiler == "tsc"),
```

Comment:
Here also, `transpiler` can be ignored if `emit_declaration_only` is True so more correct would be
```
0 if transpile_target_name != None else (1 if transpiler == "tsc" else -1)
```

Reply:
And then as a defence in depth, also fail with a friendly error message if `emit_declaration_only` is True and user sets `transpiler` since that case is not valid